### DISCUSSION TITLE
python -m textual.app not running

### DISCUSSION THREAD

**Discussion by lllama** [Textualize/textual#17] [General]
The simple.py script and running `python -m textual.app` both crash with 
```
TypeError: 'LayoutOptions' object is not subscriptable
```
on startup.

Posting here rather than in issues, as I know this is pre-release code. I'll have a prod tomorrow and update here if I find anything.

**Comment by willmcgugan** [comment]
I think I broke master. If you try the latest from Pypi it *should* work.

In the future I'm going to try and keep the main branch in a working state. If you want cutting edge code, check out the grid branch. But I can't guarantee it won't break.

**Reply by lllama** [reply]
Awesome. Grid not working for me (several imports missing/wrong/non-existent) but appreciate that this is all very much in flux atm. The `simple.py` script runs with the PyPI version though (0.1.2 at the time of writing).

I was going to have a play and see whether I could get something drawn on Windows, as vt100 and xterm support is a lot better these days, but wanted to see what "working" looked like on Linux first.

**Reply by willmcgugan** [reply]
I had thought that there would need to be a Windows driver, but now I think Windows Terminal might just handle it.

"python -m textual.app" should work in the grid branch. The API has changed since simple.py

**Reply by lllama** [reply]
![image](https://user-images.githubusercontent.com/598411/124442054-3e885900-dd74-11eb-8a98-4450eb94c5ad.png)

Looks like it will draw things well enough. I had to hack out `termios`, as that's not available on Windows. And I removed the use of `signal` as it was complaining about a missing attribute. Pretty sure those changes then broke mouse and keyboard events, as there was no interactivity, but just seeing things drawn warms my heart. 

It also looks like the emojis in the example readme broke `charmap_decode` but that could well be my git config doing something funky, as vim says that the readme has DOS line endings, which I doubt is really the case.

**Reply by willmcgugan** [reply]
That's great. Good to know that Windows compatibility isn't going to all that much work!

**Reply by lllama** [reply]
Looks like it's 80% there. Just the remaining 80% to do!
